src/content/en/updates/2019/02/intersectionobserver-v2.md

```diff
@@ -0,0 +1,226 @@
+[`DocumentOrShadowRoot.elementFromPoint()`](https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/elementFromPoint)
+and then digging deeper, the same information cannot be obtained if the element in question is
+located in a third-party iframe.
+
+## Why is actual visibility such a big deal?
+
+The Internet is, unfortunately, a place that attracts bad actors with even worse intentions.
+For example, a shady publisher that serves pay-per-click ads on a content site might be incentivized
+to trick people into clicking their ads to increase the publisher's ad payout (at least
+for a short period, until the ad network catches them).
+Typically, such ads are served in iframes.
+Now if the publisher wanted to get users to click such ads, they could make the ad iframes
+completely transparent by applying a CSS rule `iframe { opacity: 0; }` and overlaying the iframes
+on top of something attractive, like a cute cat video that users would actually want to click.
+This is called *clickjacking*.
+You can see such a clickjacking attack in action in the upper section of this
+[demo](https://trick-ad-click.glitch.me/) (try "watching" the 🐈 cat video
+and ☑️ activate "trick mode").
+You will notice that the ad in the iframe "thinks" it received legitimate clicks, even if it was
+completely transparent when you (pretendedly involuntarily) clicked it.
+
+<img src="../../images/2019/02/trick-ad-click.png" alt="Tricking a user into clicking an ad by
+    styling it transparent and overlaying it on top of something attractive"
+    intrinsicsize="1000x666">
+
+## How does Intersection Observer&nbsp;v2 fix this?
+
+Intersection Observer&nbsp;v2 introduces the concept of tracking the actual "visibility" of a target
+element as a human being would define it.
+By setting an option in the
+[`IntersectionObserver` constructor](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/IntersectionObserver),
+intersecting
+[`IntersectionObserverEntry`](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserverEntry)s
+(pardon the wrong plural ending here) will then contain a new boolean field named `isVisible`.
+A `true` value for `isVisible` is a strong guarantee from the underlying implementation
+that the target element is completely unoccluded by other content
+and has no visual effects applied that would alter or distort its display on screen.
+In contrast, a `false` value means that the implementation cannot make that guarantee.
+
+An important detail of the
+[spec](https://w3c.github.io/IntersectionObserver/v2/#calculate-visibility-algo)
+is that the implementation *is permitted* to report false negatives
+(that is, setting `isVisible` to `false` even when the target element is completely visible and
+unmodified), but it *may never* report a false positive.
+For performance or other reasons, an implementation may use an algorithm which is not 100% accurate,
+as long as all inaccuracies are false negatives.
+
```

Comment:
I'd like to see a section here titled something like "Caveats" or "Warning!"

The new section should **emphasize** that visibility is much more expensive to compute than intersection. For that reason, V2 is not intended to be used broadly in the way that V1 is; it is focused on combatting fraud. V2 functionality should be used only when V1 functionality is truly insufficient.

Reply:
I'm adding a big, red warning box.